app/src/main/java/com/conupods/OutdoorMaps/MapInitializer.java

```diff
@@ -7,18 +7,154 @@
 import android.widget.EditText;
 import android.widget.TextView;
 
+import com.conupods.IndoorMaps.IndoorBuildingOverlays;
 import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+
+import java.util.ArrayList;
+import java.util.List;
 
 import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
 import static android.view.inputmethod.EditorInfo.IME_ACTION_SEARCH;
 
-public class MapInitializer {
+public class MapInitializer{
+
+    private static final LatLng centerOfHall = new LatLng(  45.49728190486448,  	-73.57892364263535);
+    private static final LatLng centerOfJMSB = new LatLng(45.49524950613837, -73.57895582914352);
+    private static final LatLng centerOfLOYCC = new LatLng(45.45824552786007,    -73.64033281803131);
+    private static final LatLng centerOfLOYVL = new LatLng( 45.459086497919344,     -73.63828897476196);
 
-    CameraController mCameraController;
+    private CameraController mCameraController;
+    private IndoorBuildingOverlays mIndoorBuildingOverlays;
+    private OutdoorBuildingOverlays mOutdoorBuildingOverlays;
+    private GoogleMap mMap;
     private static final String TAG = "MapInitializer";
 
-    public MapInitializer(CameraController CameraController) {
-        mCameraController = CameraController;
+    public MapInitializer(CameraController cameraController, IndoorBuildingOverlays indoorBuildingOverlays, OutdoorBuildingOverlays outdoorBuildingOverlays, GoogleMap map) {
+        mCameraController = cameraController;
+        mIndoorBuildingOverlays = indoorBuildingOverlays;
+        mOutdoorBuildingOverlays = outdoorBuildingOverlays;
+        mMap = map;
+    }
+
+    public void onCameraChange(){
+
+        mMap.setOnCameraIdleListener(()->{
+
+            LatLngBounds bounds = mMap.getProjection().getVisibleRegion().latLngBounds;
+
+            if (mMap.getCameraPosition().zoom > 18.3) {
+                mOutdoorBuildingOverlays.removePolygons();
+                if (bounds.contains(centerOfJMSB)) {
+                    mIndoorBuildingOverlays.displayOverlayJMSB();
+                    mIndoorBuildingOverlays.showButtonsJMSB();
+                }
+                else if(bounds.contains(centerOfHall)) {
+                    mIndoorBuildingOverlays.displayOverlayHall();
+                    mIndoorBuildingOverlays.showButtonsHALL();
+                }
+                else if(bounds.contains(centerOfLOYCC)) {
+                    mIndoorBuildingOverlays.displayOverlayLOYCC();
+                }
+                else if(bounds.contains(centerOfLOYVL)) {
```

Comment:
you can use the chain of responsibility design pattern here, and just chain the multiple building overlays